src/data/mystery-encounters/mystery-encounter.ts

```diff
@@ -0,0 +1,963 @@
+   * MUST EVENTUALLY BE DISABLED TO CONTINUE TO NEXT WAVE
+   * Default false
+   */
+  continuousEncounter: boolean;
+  /**
+   * Maximum number of times the encounter can be seen per run
+   * Rogue tier encounters default to 1, others default to 3
+   */
+  maxAllowedEncounters: number;
+  /**
+   * If true, encounter will not animate the target Pokemon as part of battle animations
+   * Used for encounters where it is not a "real" battle, but still uses battle animations and commands (see {@linkcode FunAndGamesEncounter} for an example)
+   */
+  hasBattleAnimationsWithoutTargets: boolean;
+  /**
+   * If true, will skip enemy pokemon turns during battle for the encounter
+   * Used for encounters where it is not a "real" battle, but still uses battle animations and commands (see {@linkcode FunAndGamesEncounter} for an example)
+   */
+  skipEnemyBattleTurns: boolean;
+  /**
+   * If true, will skip COMMAND input and go straight to FIGHT (move select) input menu
+   */
+  skipToFightInput: boolean;
+
+  /**
+   * Event callback functions
+   */
+  /** Event when Encounter is first loaded, use it for data conditioning */
+  onInit?: (scene: BattleScene) => boolean;
+  /** Event when battlefield visuals have finished sliding in and the encounter dialogue begins */
+  onVisualsStart?: (scene: BattleScene) => boolean;
+  /** Event triggered prior to {@linkcode CommandPhase}, during {@linkcode TurnInitPhase} */
+  onTurnStart?: (scene: BattleScene) => boolean;
+  /** Event prior to any rewards logic in {@linkcode MysteryEncounterRewardsPhase} */
+  onRewards?: (scene: BattleScene) => Promise<void>;
+  /** Will provide the player party EXP before rewards are displayed for that wave */
+  doEncounterExp?: (scene: BattleScene) => boolean;
+  /** Will provide the player a rewards shop for that wave */
+  doEncounterRewards?: (scene: BattleScene) => boolean;
+  /** Will execute callback during VictoryPhase of a continuousEncounter */
+  doContinueEncounter?: (scene: BattleScene) => Promise<void>;
+
+  /**
+   * Requirements
+   */
+  requirements: EncounterSceneRequirement[];
+  /** Primary Pokemon is a single pokemon randomly selected from the party that meet ALL primary pokemon requirements */
+  primaryPokemonRequirements: EncounterPokemonRequirement[];
+  /**
+   * Secondary Pokemon are pokemon that meet ALL secondary pokemon requirements
+   * Note that an individual requirement may require multiple pokemon, but the resulting pokemon after all secondary requirements are met may be lower than expected
+   * If the primary pokemon and secondary pokemon are the same and ExcludePrimaryFromSupportRequirements flag is true, primary pokemon may be promoted from secondary pool
+   */
+  secondaryPokemonRequirements: EncounterPokemonRequirement[];
+  excludePrimaryFromSupportRequirements: boolean;
+  primaryPokemon?: PlayerPokemon;
+  secondaryPokemon?: PlayerPokemon[];
+
+  /**
+   * Post-construct / Auto-populated params
+   */
```

Comment:
you could use 
```suggestion
  // #region Post-construct / Auto-populated params
```

Make sure to close it with
```ts
  // #endregion
```

This would show up like this on vs-codium minimap:
<img width="174" alt="image" src="https://github.com/user-attachments/assets/6e510219-209f-47ae-aa44-b3daf46b1c83">